test/PushNotificationTests/APITests.cpp

```diff
@@ -0,0 +1,184 @@
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(ChannelRequestUsingRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ChannelRequestUsingRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingMultipleRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingMultipleRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(ThreeChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ThreeChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(BackgroundActivation)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+            auto LocalBackgroundTask = winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(c_comServerId, CLSCTX_ALL);
+            auto mockBackgroundTaskInstance = winrt::make<MockBackgroundTaskInstance>();
```

Comment:
Can you also please verify that the payload that you pass into the Mock here is the same payload that is extracted when we launch the app?

Reply:
Yes. This is already being checked in the ExtendedActivationKind::Push block. I compare the the payload from PushNotificationReceivedEventArgs with the c_rawNotificationPayload listed in TestDef.h